docs: add universal moderators handbook (#39417)
freeCodeCamp/freeCodeCamp: docs/flight-manuals/moderator-handbook.md

Before merging any pull request, make sure that GitHub has green checkmarks for everything. If there are any X's, investigate them first and figure out how to get them turned into green checkmarks first.

Sometimes there will be a Merge Conflict. This means that another pull request has made a change to that exact same part of that same file. GitHub has a tool for addressing these merge conflicts right on GitHub. You can try to address these conflicts. Just use your best judgement. The pull request's changes will be on top, and the Master branch's changes will be on bottom. Sometimes there will be redundant information in there that can be deleted. Before you finish, be sure to delete the `<<<<<<`, `======`, and `>>>>>>` that Git adds to indicate areas of conflict.

If the pull request looks ready to merge (and doesn't require approval from @raisedadead), you can go ahead and merge it. Be sure to use the default "Squash and Merge" functionality on GitHub. This will squash all the pull requests commits down into a single commit, which makes the Git history much easier to read.

You should then comment on the pull request, thanking the contributor in your own personal way.

> If the author of the pull request is a "first time contributor" you should also congratulate them on their first merged pull request to the repository. Here's how you can tell that:

Reviewer: ```suggestion
If the author of the pull request is a "first time contributor" you should also congratulate them on their first merged pull request to the repository. You can look at the upper right-hand corner of the PR's body to determine a first-time contributor.  It will show `First-time contributor` as shown below:
```

- If it's a simple change to an existing challenge, flag as `help wanted` and, optionally, as `first-timers-only`. Use other tags as appropriate.
- If the issue is more significant, flag as `bug`.
  &nbsp;
  If there is any ambiguity as to the proper course of action on an issue, feel free to tag @raisedadead on the issue get his opinion on it, then add the `Discussing` label.

3. **Duplicate Issues** If an issue is the same as another reported issue, the prior reported issue should take precedence. Flag as `Duplicate`, paste the following message replacing `#XXXXX` with the issue number, then close the issue.

   > Thank you for reporting this issue.
   >
   > This is a standard message notifying you that this issue appears to be very similar to issue #XXXXX, so I am closing it as a duplicate.
   >
   > If you think I'm wrong in closing this issue, please reopen it and add further clarification. Thank you and happy coding.

> 4. **Fixed in staging** Some problems may have already been fixed in staging, but don't have an GitHub issue associated with them. If this is the case, you can paste the following message, then close the issue:

Reviewer: ```suggestion
4. **Fixed in staging** Some problems may have already been fixed in staging, but don't have an GitHub issue associated with them. If this is the case, you can paste the following message, close the issue, and add a `status: resolved/shipping` label:
```

   > Thank you for reporting this issue.
   >
   > This is a standard message notifying you that the problem you mentioned here is present in production, but that it has already been fixed in staging. This means that the next time we push our staging branch to production, this problem should be fixed. Because of this, I'm closing this issue.
   >
   > If you think I'm wrong in closing this issue, please reopen it and add further clarification. Thank you and happy coding.

### Closing Stale, Outdated, Inactive Issues and Pull Requests

> - Merge-conflicts and Stale PRs that have not seen any activity from anyone for 21 days (3 weeks from the last activity), we can close them in an automated/bot script or the moderators themselves.

Reviewer: ```suggestion
- Stale Issues or PRs are those that have not seen any activity from the OP for 21 days (3 weeks from the last activity), but only after a moderator has requested more information/changes.  These can be closed in an automated/bot script or by the moderators themselves.
```

- Activity is defined as: Comments requesting an update on the PR and triages like `status: update needed` label etc.

> - If the OP ask for additional assistance or even time, the above can be relaxed and revisited after a response is given. In any case the mods should use their best judgement to resolve the outstanding PR's status.

Reviewer 2: ```suggestion
- If the OP asks for additional assistance or even time, the above can be relaxed and revisited after a response is given. In any case the mods should use their best judgement to resolve the outstanding PR's status.
```

Note that once you are a moderator on the forum, you will start to see blue moderator hints about forum members, like "this is the first time [person] has posted - let's welcome them to the community!" or "[person] hasn't posted in a long time - let's welcome them back."

![A blue text message saying "this is the first time [person] has posted - let's welcome them to the community!](https://i.imgur.com/mPmVgzK.png)

These are opportunities for you to welcome them and make them feel extra special. You never know which person who's marginally involved may become our next super-helper, helping many other people in their coding journey. Even the smallest kindness may trigger a cascade of good deeds.

### Deleting forum posts

Forum moderators have the ability to delete user's posts. You should only do this for the following instances:

1. Someone has posted a pornographic or graphically violent image.
2. Someone has posted a link or code that is malicious in nature, and could harm other campers who click on it.
3. Someone has flooded a thread with lots of spam messages.

### Dealing with spam

> For the first spam post of a user, send them a message explaining the problem, and remove the link or post as appropriate. Leave a note on the user's profile explaining the action you have taken. If the problem persists, then follow the process above. Quietly block user from posting, then send warning with Code of Conduct. Check the box in the private message indicating that your message is a "formal warning."

Reviewer 2: ```suggestion
For the first spam post of a user, send them a message explaining the problem, and remove the link or post as appropriate. Leave a note on the user's profile explaining the action you have taken. If the problem persists, then follow the process above. Quietly block the user from posting (using the silence option on the User Admin panel), then send a warning with the Code of Conduct. Check the box in the private message indicating that your message is a "formal warning."
```

You can ask questions and report incidents in the in the [staff forum section](https://forum.freecodecamp.com/c/staff).

### Dealing with off-topic conversations

> Posts or topics that seems to be in the wrong place, can be re-categorized or renamed to whatever would be appropriate.

Reviewer 2: ```suggestion
Posts or topics that seems to be in the wrong place can be re-categorized or renamed to whatever would be appropriate.
```

- If you decide to ban someone, it means they're unwilling to abide to our Code of Conduct. Therefore unbanning a Camper should rarely occur. However, if the need arises, you can do so by clicking on the server name, choosing "Server Settings", choosing "Bans", selecting the user you wish to unban, and clicking "Revoke Ban".

Discord Bans are global - you cannot ban a user from a specific channel, only from the entire server.

4. **Deleting messages**
   Moderators have the ability to delete messages on Discord. They should only exercise this ability in four very specific situations:

- Someone has posted a pornographic or graphically violent image.
- Someone has posted a link or code that is malicious in nature, and could harm other campers who click on it.
- Someone has flooded the chat with lots of spam messages to such an extreme extent (usually involving bots) as to render chat completely unusable.
- Someone has posted advertisement and / or a self-promoting message / image (social media).

> In all other situations - even situations where the code of conduct is violated - Moderators should not delete the message as these are an important historic record. When you do delete a message, make sure you _always_ make a screenshot of it first! The screenshot can be logged in the #mod-log channel, but for the #admin log it is sufficient to say the evidence was "removed due to sensitive content".

Reviewer 2: ```suggestion
In all other situations - even situations where the code of conduct is violated - Moderators should not delete the message as these are an important historic record. When you do delete a message, make sure you take a screenshot of it first! The screenshot can be logged in the #mod-log channel, but for the #activity-log it is sufficient to say the evidence was "removed due to sensitive content". Note: If the message contains material that would be illegal to take a screenshot of, copy the message link instead - provide that message link to @raisedadead to forward to Discord's Trust and Safety team.
```

5. **Don’t use @everyone or @here**
   Don’t use @everyone or @here under any circumstances! Every single person in that chat room will get a notification. In some cases, tens of thousands of people.
   Instead, if you want people to see an announcement, you can pin it to the channel to allow everyone to read it.

6. **Don’t threaten to ban or suspend**

>    If a camper is breaking the code of conduct, don’t threaten to ban or suspend them, and never warn them in public. Instead, talk to them privately, or send them a DM and issue a suspension (per the above protocol). No one else in that channel needs to know that you banned / suspended the person - campers can view the summary in the #admin channel if they want to keep up on that information. If a violation was clearly unintended and doesn't warrant a suspension or private conversation, make the offending camper aware of his / her actions without making it come across as a warning. For example:

Reviewer 2: ```suggestion
   If a camper is breaking the code of conduct, don’t threaten to ban or suspend them, and never warn them in public. Instead, talk to them privately, or send them a DM and issue a suspension (per the above protocol). No one else in that channel needs to know that you banned / suspended the person - campers can view the summary in the #activity-log channel if they want to keep up on that information. If a violation was clearly unintended and doesn't warrant a suspension or private conversation, make the offending camper aware of his / her actions without making it come across as a warning. For example:
```

# How we retire inactive moderators

Please note that we will frequently remove mods whom we think are inactive. When we do this we will send the following message:

> This is a standard message notifying you that, since you don't seem to have been an active moderator recently, we're removing you from our Moderator team. We deeply appreciate your help in the past.

> If you think we did this in error, or once you're ready to come back and contribute more, just reply to this message letting me know.

# How our Contributors room works

Anyone is welcome in the [Contributors room on our Discord](https://discord.gg/KVUmVXA). It is the designated chat room for moderators and other campers who are contributing to our community in any number of ways, including through study groups.

Our assumption is that contributors will read anything in this room that directly mentions them with an `@username`. Everything else is optional. But feel free to read anything anyone posts in there and interact.

# Dealing with solicitors

> You may be approached by organizations who want to partner or co-brand with freeCodeCamp in some way. Once you realize that this is what they're after, please stop talking to them and tell them to email quincy@freecodecamp.org. He get proposals like this all the time and is in the best position to judge whether such a relationship will be worth it for our community (and it rarely is).

Reviewer 2: ```suggestion
You may be approached by organizations who want to partner or co-brand with freeCodeCamp in some way. Once you realize that this is what they're after, please stop talking to them and tell them to email quincy@freecodecamp.org. He gets proposals like this all the time and is in the best position to judge whether such a relationship will be worth it for our community (and it rarely is).
```

There are also situations where the contributor is trying to add a link to their own website, or include a library they themselves created, or has a frivolous edit that doesn't serve to help anyone but themselves.

In both of these situations, you should go ahead and close their pull request and reply with this standard message:

> Thank you for opening this pull request.
>
> This is a standard message notifying you that we've reviewed your pull request and have decided not to merge it. We would welcome future pull requests from you.
>
> Thank you and happy coding.

If you need a second opinion on a pull request, go ahead and leave your comments on the pull request, then add the "discussing" label to the pull request.

## Moderating GitHub Issues

freeCodeCamp is an active open source project. We get new issues every day, all of which need to be triaged and labeled.

### Types of GitHub Issues

1. **Code Help Requests**, which people have mistakenly created GitHub issues for. If someone is asking for help, paste the following message, then close the issue.

   > Thank you for reporting this issue.

>    >
>    > This is a standard message notifying you that this issue seems to be a request for help. Instead of asking for help here, please click the \*\*"Help"\*\* button on the challenge on freeCodeCamp, which will help you create a question in the right part of the forum. Volunteers on the forum usually respond to questions within a few hours and can help determine if there is an issue with your code or the challenge's tests.
>    >
> If the forum members determine there is nothing wrong with your code, you can request this issue to be reopened. 
> 
> Thank you and happy coding.
> 

Reviewer 2: ```suggestion
   >
   > This is a standard message notifying you that this issue seems to be a request for help. Instead of asking for help here, please click the \*\*"Help"\*\* button on the challenge on freeCodeCamp, which will help you create a question in the right part of the forum. Volunteers on the forum usually respond to questions within a few hours and can help determine if there is an issue with your code or the challenge's tests.
   >
   > If the forum members determine there is nothing wrong with your code, you can request this issue to be reopened. 
   > 
   > Thank you and happy coding.
   
```
Looks like these lines fell out of the blockquote.

- If it's a simple change to an existing challenge, label as `first timers only`, otherwise label as `help wanted`. Use other labels as appropriate.
- If the issue is more significant, flag as `bug`.
  &nbsp;
  If there is any ambiguity as to the proper course of action on an issue, feel free to tag @raisedadead on the issue get his opinion on it, then add the `Discussing` label.

3. **Duplicate Issues** If an issue is the same as another reported issue, the prior reported issue should take precedence. Flag as `Duplicate`, paste the following message replacing `#XXXXX` with the issue number, then close the issue.

   > Thank you for reporting this issue.
   >
   > This is a standard message notifying you that this issue appears to be very similar to issue #XXXXX, so I am closing it as a duplicate.
   >
   > If you think I'm wrong in closing this issue, please reopen it and add further clarification. Thank you and happy coding.

> 4. **Fixed in staging** Some problems may have already been fixed in staging, but don't have an GitHub issue associated with them. If this is the case, you can paste the following message, close the issue, and add a `status: resolved/shipping` label:

Reviewer 3: ```suggestion
4. **Fixed in staging** Some problems may have already been fixed in staging, but don't have a GitHub issue associated with them. If this is the case, you can paste the following message, close the issue, and add a `status: resolved/shipping` label:
```
